src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -885,6 +889,176 @@ MavlinkReceiver::handle_message_set_position_target_local_ned(mavlink_message_t
+		offboard_control_mode.ignore_bodyrate_x = (bool)(set_position_target_global_int.type_mask & 0x800);
+		offboard_control_mode.ignore_bodyrate_y = (bool)(set_position_target_global_int.type_mask & 0x800);
+		offboard_control_mode.ignore_bodyrate_z = (bool)(set_position_target_global_int.type_mask & 0x800);
+
+		/* yaw ignore flag mapps to ignore_attitude */
+		bool is_force_sp = (bool)(set_position_target_global_int.type_mask & (1 << 9));
+
+		bool is_takeoff_sp = (bool)(set_position_target_global_int.type_mask & 0x1000);
+		bool is_land_sp = (bool)(set_position_target_global_int.type_mask & 0x2000);
+		bool is_loiter_sp = (bool)(set_position_target_global_int.type_mask & 0x3000);
+		bool is_idle_sp = (bool)(set_position_target_global_int.type_mask & 0x4000);
+
+		offboard_control_mode.timestamp = hrt_absolute_time();
+		_offboard_control_mode_pub.publish(offboard_control_mode);
+
+		/* If we are in offboard control mode and offboard control loop through is enabled
+		 * also publish the setpoint topic which is read by the controller */
+		if (_mavlink->get_forward_externalsp()) {
+
+			vehicle_control_mode_s control_mode{};
+			_control_mode_sub.copy(&control_mode);
+
+			if (control_mode.flag_control_offboard_enabled) {
+				if (is_force_sp && offboard_control_mode.ignore_position &&
+				    offboard_control_mode.ignore_velocity) {
+
+					PX4_WARN("force setpoint not supported");
+
+				} else {
+					/* It's not a pure force setpoint: publish to setpoint triplet  topic */
+					position_setpoint_triplet_s pos_sp_triplet{};
+
+					pos_sp_triplet.timestamp = hrt_absolute_time();
+					pos_sp_triplet.previous.valid = false;
+					pos_sp_triplet.next.valid = false;
+					pos_sp_triplet.current.valid = true;
+
+					/* Order of statements matters. Takeoff can override loiter.
+					 * See https://github.com/mavlink/mavlink/pull/670 for a broader conversation. */
+					if (is_loiter_sp) {
+						pos_sp_triplet.current.type = position_setpoint_s::SETPOINT_TYPE_LOITER;
+
+					} else if (is_takeoff_sp) {
+						pos_sp_triplet.current.type = position_setpoint_s::SETPOINT_TYPE_TAKEOFF;
+
+					} else if (is_land_sp) {
+						pos_sp_triplet.current.type = position_setpoint_s::SETPOINT_TYPE_LAND;
+
+					} else if (is_idle_sp) {
+						pos_sp_triplet.current.type = position_setpoint_s::SETPOINT_TYPE_IDLE;
+
+					} else {
+						pos_sp_triplet.current.type = position_setpoint_s::SETPOINT_TYPE_POSITION;
+					}
+
+					/* set the local pos values */
+					vehicle_local_position_s local_pos{};
+
+					if (!offboard_control_mode.ignore_position && _vehicle_local_position_sub.copy(&local_pos)) {
+						if (!globallocalconverter_initialized()) {
+							globallocalconverter_init(local_pos.ref_lat, local_pos.ref_lon,
+										  local_pos.ref_alt, local_pos.ref_timestamp);
+							pos_sp_triplet.current.position_valid = false;
+
+						} else {
+							globallocalconverter_tolocal(set_position_target_global_int.lat_int / 10000000.0f,
```

Comment:
The first  param of `globallocalconverter_tolocal` is a `double` so you should do the division with `10000000.0` (without `f`), or just use `1e7`. That way the calculation is done in double precision which can be required for lat/lon.